### PR TITLE
SW-4956 Integrate file upload with BE

### PR DESCRIPTION
**Pull request by karthikbtf**
- added types/redux/service
- updated component
- also added roles/guard for deliverable data

### CODE REVIEW

**Comment by karthikbtf** [comment]
This works not but needs BE PR https://github.com/terraware/terraware-server/pull/1716

**Comment by karthikbtf** [comment]
> This works now but needs BE PR [terraware/terraware-server#1716](https://github.com/terraware/terraware-server/pull/1716)

typo, works "now"

File: src/services/DeliverablesService.ts [hunk 1]
```diff
@@ -84,10 +89,31 @@ const list = async (
...
+    };
+    const entity = { description, file };
+
+    return httpDocumentUpload.post({ urlReplacements, entity, headers });
+  });
+
+  const results = await getPromisesResponse<Response>(promises);
+  return results.every((result) => result !== null && result.requestSucceeded === true);
+};
+
```
Comment (joshbuchea) [inline src/services/DeliverablesService.ts:111]: How do we handle a partial failure? (one or more document uploads completed successfully, while one or more documents uploads failed)

**Review by joshbuchea — APPROVED** [review]

File: src/services/DeliverablesService.ts (hunk 1 above)
Comment (karthikbtf) [reply src/services/DeliverablesService.ts:111]: We don't have this pattern today, something we can add later. For now if any one fails, we show an error but we refresh the documents list so user knows which ones succeeded.

**Pull request merged**